src/irmin-pack/unix/stats.mli

```diff
@@ -67,29 +67,82 @@ module Index : sig
   val export : stat -> t
 end
 
-type t = { pack_store : Pack_store.stat; index : Index.stat }
+module File_manager_stats : sig
+  type t = private {
+    mutable dict_flushes : int;
+    mutable suffix_flushes : int;
+    mutable index_flushes : int;
+    mutable auto_dict: int;
+    mutable auto_suffix: int;
+    mutable auto_index: int;
+    mutable flush : int
+  }
+  [@@deriving irmin]
+
+  type stat
+
+  val export : stat -> t
+end
+
+type t = {
+  pack_store : Pack_store.stat;
+  index : Index.stat;
+  file_manager : File_manager_stats.stat;
+}
+(** Record type for all statistics that will be collected. There is a single
+    instance (the "get instance") which is returned by {!get} below. *)
 
 val reset_stats : unit -> unit
+(** [reset_stats ()] will call the relevant [clear] function on each field of
+    "get instance". This typically resets the fields (e.g. to 0 for an int
+    field). *)
 
 val get : unit -> t
-(** [get ()] returns the {!t} that stores the satistics. If {!report_pack_store}
-    or {!report_index} is not called before, the content will be filled with
-    default value, decided at create time (most the time, [0]). *)
+(** [get ()] returns the {!t} that stores the satistics (the "get instance"). If
+    {!report_pack_store} or {!report_index} is not called before, the content
+    will be filled with default value, decided at create time (most the time,
+    [0]). *)
 
 val report_pack_store : field:Pack_store.field -> unit
-(** [report_pack_store ~field] incrementes the [field] value in the [pack_store]
-    stats. It also incrementes the [total] field in {!Pack_store.t} when the
+(** [report_pack_store ~field] increments the [field] value in the [pack_store]
+    stats. It also increments the [total] field in {!Pack_store.t} when the
     field is related to [finds]. *)
 
 val report_index : unit -> unit
 (** [report_index ()] fills the [stats] with value from the {!Index.Stats}
-    module. *)
+    module. This essentially copies the "current" values from {!Index.Stats} to
+    the [get()] instance [index] field. *)
 
 val incr_appended_hashes : unit -> unit
+(** [incr_appended_hashes ()] increments the field [appended_hashes] for
+    [pack_store] in the "get instance". *)
```

Comment:
I think that all the information in your docstring is contained in the name of the function. This is the typical pattern in which a docstring is not necessary

Reply:
I wanted to emphasize that these functions are tied to the single instance of type t (it wasn't obvious to me before reading the .ml that this was the case... and ideally I like .mlis to be understood without reading the .ml). Hope this is OK.